addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/ContentSecurityPolicyMissingScannerUnitTest.java

```diff
@@ -0,0 +1,166 @@
+    private static final String HEADER_TEXT = "Content-Type: text/plain";
+    private static final String HEADER_CSP = "Content-Security-Policy: x";
+    private static final String HEADER_X_CSP = "X-Content-Security-Policy: x";
+    private static final String HEADER_WEBKIT_CSP = "X-WebKit-CSP: x";
+    private static final String HEADER_REPORT_ONLY = "Content-Security-Policy-Report-Only: x";
+
+    @Override
+    protected ContentSecurityPolicyMissingScanner createScanner() {
+        return new ContentSecurityPolicyMissingScanner();
+    }
+
+    @Test
+    public void givenMissingCspHeaderThenAlertRaised() throws Exception {
+        // Given
+        HttpMessage msg = createHttpMessageWithHeaders(new String[] {HEADER_HTML});
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertContentSecurityPolicyAlertRaised();
+    }
+
+    @Test
+    public void givenOnlyCspHeaderAtMediumAlertThresholdThenNoAlertRaised() throws Exception {
+        // Given
+        rule.setAlertThreshold(Plugin.AlertThreshold.MEDIUM);
+        HttpMessage msg = createHttpMessageWithHeaders(new String[] {HEADER_HTML, HEADER_CSP});
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), is(0));
+    }
+
+    @Test
+    public void givenOnlyCspHeaderAtLowAlertThresholdThenAlertRaised() throws Exception {
+        // Given
+        rule.setAlertThreshold(Plugin.AlertThreshold.LOW);
+        HttpMessage msg = createHttpMessageWithHeaders(new String[] {HEADER_HTML, HEADER_CSP});
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertContentSecurityPolicyAlertRaised();
+    }
+
+    @Test
+    public void givenAllCspHeadersAtLowAlertThresholdThenNoAlertRaised() throws Exception {
+        // Given
+        rule.setAlertThreshold(Plugin.AlertThreshold.LOW);
+        String[] headers = {HEADER_HTML, HEADER_CSP, HEADER_X_CSP, HEADER_WEBKIT_CSP};
+        HttpMessage msg = createHttpMessageWithHeaders(headers);
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), is(0));
+    }
+
+    @Test
+    public void givenTextContentTypeAtMediumAlertThresholdThenNoAlertRaised() throws Exception {
+        // Given
+        rule.setAlertThreshold(Plugin.AlertThreshold.MEDIUM);
+        HttpMessage msg = createHttpMessageWithHeaders(new String[] {HEADER_TEXT});
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), is(0));
+    }
+
+    @Test
+    public void givenTextContentTypeAtLowAlertThresholdThenAlertRaised() throws Exception {
+        // Given
+        rule.setAlertThreshold(Plugin.AlertThreshold.LOW);
+        HttpMessage msg = createHttpMessageWithHeaders(new String[] {HEADER_TEXT});
```

Comment:
Should also test that the alert is not raised when it has the CSP header (ideally it should have the same tests as HTML).